packages/gatsby-theme-docs/src/plugins/remark-mdx-mermaid.js

```diff
@@ -0,0 +1,18 @@
+const visit = require('unist-util-visit');
+
+/**
+ * Given the MDXAST ast, look for all fenced codeblocks that have a language of
+ * `mermaid` and replace them with a <Mermaid> JSX component containing the mermaid code.
+ * MDXAST reference: https://github.com/mdx-js/specification#jsx-1
+ *
+ * @param {object}  ast
+ * @return {function}
+ */
+module.exports = () => (ast) => {
+  visit(ast, { type: 'code', lang: 'mermaid' }, (node) => {
+    return Object.assign(node, {
+      type: 'jsx',
+      value: '<Mermaid graph={`' + node.value + '`} />',
```

Comment:
Passing the graph as children body content instead of as a prop caused various encoding and robustness issues. so it's passed as a prop now.